### PR TITLE
Game_logic

### PR DESCRIPTION
**Pull request by vmwhoami**
Implemented the Tic Tac Toe game logic using classes and methods.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class Board
+  attr_accessor :highest_turns
```
Comment (Dandush03) [inline lib/board.rb:2]: make sure that you most use write/read as far as I've seen in your code you only need write

**Review by Dandush03 — COMMENTED** [review]
# Status: 🔴 Required Changes 🔴 

## Great work for the first attempt. but there are some issues we need to fix in order to continue to the next milestone.

- [ ] When giving an invalid move it's not accepting the input (this is great) and also it jump to the next player move (this is bad) 
> I was able to win this game without playing a single time with the second player
> Capture
- [ ] Please avoid the use of attribute accessor unless needed

[__OPTIONAL__]

#### Please remove the sleep function when displaying the rules, since if it's an experienced user it will be a little bit annoying to wait till all of them load to play the game, or add a way to skip it

### Please make sure to fix does issue before submitting for another review, I will also going to leave you some comments on your code. please make sure to go through them as well 

## Happy Coding! 🤓 💻

**Comment by lekegitrepo** [comment]
#### Hi @vmwhoami and @somoye123 :clap: :clap:
#### You have done a good so far, however, the game is outputting an error whenever I type the player input. Check the image :point_down:  below.

Screenshot from 2020-07-15 02-34-16

**_Fix this and submit for another review._**

#### _Reviewed by Temitayo Adeleke_

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
#### Status: Request changes

**Review by trekab — CHANGES_REQUESTED** [review]
## Changes Required! :recycle: 
Great job Team! Your game has one issue you need to resolve before it is approved. When players enter the same name or a blank name, the logic of your game breaks. Please restrict the players not to input the same user name and not to submit an empty user name.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, congratulations on fixing all the comments made by the previous TSE's :100: We have now met all the minimum requirements :+1: 

Go ahead and merge into development :slightly_smiling_face: 
Good luck with your next milestone :fireworks: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**